File: Python_CV/session 4/task2.py
import cv2
from matplotlib import pyplot as plt
import numpy as np

Img = cv2.imread('image/34.jpg')
gray = cv2.cvtColor(Img, cv2.COLOR_BGR2GRAY)
[h, w] = gray.shape
sigma = 10
print(int(np.round(sigma * np.random.rand(h, w))))
# H1 = np.ones((3, 3), dtype=float)/9
# H2 = np.ones((5, 5), dtype=float)/25
# H3 = np.ones((9, 9), dtype=float)/81
# I1 = cv2.filter2D(gray, ddepth=-1, kernel=H1)
# cv2.imshow('Original', gray)
# cv2.imshow('Identity', I1)
cv2.waitKey(0)

File: Python_CV/session 4/task2.m
img = rgb2gray(imread('image\12.jpg'));
[h, w] = size(img);
img = double(img) + 20 * randn(h,w);
subplot(2,2,1)
imshow(img,[])
H1 = ones(3,3)/9;
H2 = ones(5,5)/25;
H3 = ones(9,9)/81;
I1 = conv2(img, H1,'same');
[h1, w1] = size(I1)
I2 = conv2(img, H2,'same');
I3 = conv2(img, H3,'same');
subplot(2,2,2)
imshow(I1,[]);
subplot(2,2,3)
imshow(I2,[]);
subplot(2,2,4)
imshow(I3,[]);